### PR TITLE
[FLINK-24715][connectors][filesystems][formats] Update multiple Jackson dependencies to v2.13.0

### PR DESCRIPTION
**Pull request by MartijnVisser**
## What is the purpose of the change

* Update multiple Jackson dependencies to the latest version

## Brief change log

* Updated POM file
* Excluded module-info.class from Table API JAR

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (**yes** / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / **don't know**)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-planner/pom.xml [hunk 1]
```diff
@@ -356,6 +356,7 @@ under the License.
 										<exclude>META-INF/*.DSA</exclude>
 										<exclude>META-INF/*.RSA</exclude>
 										<exclude>META-INF/services/java.sql.Driver</exclude>
+										<exclude>META-INF/versions/11/module-info.class</exclude>
```
Comment (fapaul) [inline flink-table/flink-table-planner/pom.xml:359]: Can you explain this change?

**Review by fapaul — CHANGES_REQUESTED** [review]
I still see Jackson 2.4.0 in the HBase SQL connector NOTICE files is this also affected by this dependency change?

**Comment by fapaul** [comment]
Can you also check this failing test it might be that our elastic version does not play nice with this new Jackson version?

**Comment by MartijnVisser** [comment]
HBase uses org.apache.htrace:htrace-core4 which is bundled with Jackson 2.4.0

File: flink-table/flink-table-planner/pom.xml (hunk 1 above)
Comment (MartijnVisser) [reply flink-table/flink-table-planner/pom.xml:359]: The new Jackson is available as a multi-release JAR, which is why there's a separate class for Java 11. The Flink shaded tests `test_table_shaded_dependencies.sh` checks if any .*class files exist to get a clean uber JAR. That's why I'm excluding it when creating the uber JAR.

**Comment by MartijnVisser** [comment]
The test succeeded on my local Azure CI. I'll rebase to be sure and push it once more.

**Comment by MartijnVisser** [comment]
@flinkbot run azure

**Review by fapaul — APPROVED** [review]
Thanks for answering my questions :) 

I checked locally that no other version of Jackson is on the classpath than 2.13.0 and all NOTICE files bundling it are updated, Ignoring the HBase connector.

**Comment by MartijnVisser** [comment]
@AHeise Please merge

**Pull request merged**